broker.go

```diff
@@ -675,69 +716,86 @@ func (b *Broker) muCloseDeadConnection(conn *connection) {
 // offset will return offset value for given partition. Use timems to specify
 // which offset value should be returned.
 func (b *Broker) offset(topic string, partition int32, timems int64) (offset int64, err error) {
-	conn, err := b.muLeaderConnection(topic, partition)
-	if err != nil {
-		return 0, err
-	}
-	resp, err := conn.Offset(&proto.OffsetReq{
-		ClientID:  b.conf.ClientID,
-		ReplicaID: -1, // any client
-		Topics: []proto.OffsetReqTopic{
-			{
-				Name: topic,
-				Partitions: []proto.OffsetReqPartition{
-					{
-						ID:         partition,
-						TimeMs:     timems,
-						MaxOffsets: 2,
+	var resErr error
+	for retry := 0; retry < b.conf.RetryErrLimit; retry++ {
```

Comment:
Would this not be better if it was abstracted in the `connection`, so that all the requests to kafka have some sort of retry mechanism? I would also like to do some sort of exponential backoff for retries.